include/tvm/layout.h

```diff
@@ -0,0 +1,335 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file tvm/layout.h
+ * \brief Layout expression to describe the data organization of a tensor.
+ *  And BijectiveLayout to mapping two data layouts between each other.
+ */
+#ifndef TVM_LAYOUT_H_
```

Comment:
How about we rename layout.h ->data_layout just to make the header name clear, we do not have to rename anything in the class